homework/index.js

```diff
@@ -1,47 +1,136 @@
 'use strict';
 
-{
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+function renderError(error) {
+  const h2 = document.createElement('h2');
+  document.querySelector('#root').appendChild(h2);
+  h2.innerHTML = error;
+}
+async function fetchJSON(url) {
+  try {
+    const response = await fetch(url);
+    const json = await response.json();
+    return json;
```

Comment:
Excellent that you have replaced `XMLHttpRequest` with the newer `fetch()` function that directly works with promises. You could combine lines 11 & 12 into a single `return response.json()`. Explanation: A function marked with the `async` keyword always returns a promise. When you call `fetchJSON()`, for instance in line 121 of function `main()` you are already `await`-ing the result. There is no need to `await` it in `fetchSJON()` itself. Just return the promise from `response.json()` and the `await` done by the caller of `fetchJSON()` will `await` its result.